linen_examples/pixelcnn/model_test.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2020 The Flax Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+"""Tests for PixelCNN Modules."""
+
+import pixelcnn
+from flax import linen as nn
+from absl.testing import absltest
+from absl.testing import parameterized
+
+import numpy.testing as onp_testing
+
+from jax import random
+import jax.numpy as np
+from jax.config import config
+config.enable_omnistaging()
+
+
+class ModelTest(absltest.TestCase):
+
+  def setUp(self):
+    super(absltest.TestCase, self).setUp()
+    self.rng = random.PRNGKey(0)
+    self.x = np.arange(24).reshape(1, 4, 3, 2)
+
+
+  def get_weightnorm(self, params):
+    return [params[k] for k in ('direction', 'scale', 'bias')]
+
+
+  def assert_mean_and_variance(self, out):
+    # Weightnorm should ensure that, at initialization time, the outputs of the
+    # module have mean 0 and variance 1 over the non-feature dimensions.
+    onp_testing.assert_allclose(np.mean(out, (0, 1, 2)), 0., atol=1e-5)
+    onp_testing.assert_allclose(np.var(out, (0, 1, 2)), 1., atol=1e-5)
+
+
+  def test_conv(self):
+    model = pixelcnn.ConvWeightNorm(features=4, kernel_size=(3, 2))
+    out, params = model.init_with_output(self.rng, self.x)
```

Comment:
`params` should probably be named `variables`

Reply:
Ok, updated everywhere.